Modules/posixmodule.c

```diff
@@ -5612,15 +5613,17 @@ os_sched_setaffinity_impl(PyObject *module, pid_t pid, PyObject *mask)
             Py_DECREF(item);
             goto error;
         }
-        cpu = PyLong_AsLong(item);
+        cpu = PyLong_AsLongAndOverflow(item, &overflow);
+        /* PyLong_Check(item) is true, so it is guaranteed that
+           no error occurred in PyLong_AsLongAndOverflow. */
+        assert(!(cpu == -1 && PyErr_Occurred()));
         Py_DECREF(item);
-        if (cpu < 0) {
-            if (!PyErr_Occurred())
-                PyErr_SetString(PyExc_ValueError, "negative CPU number");
+        if (overflow == 1 || cpu > INT_MAX - 1) {
+            PyErr_SetString(PyExc_OverflowError, "CPU number too large");
```

Comment:
because i check whether `cpu < 0` in the `negative` case.
this check is true also in case `overflow == 1`.
would changing
`if (overflow == -1 || cpu < 0) {` 
to
`if (overflow == -1 || (assert(!overflow), cpu < 0)) {` 
make it more clear?